keras_contrib/applications/densenet.py

```diff
@@ -613,21 +656,26 @@ def __create_fcn_dense_net(nb_classes, img_input, include_top, nb_dense_block=5,
     for block_idx in range(nb_dense_block):
         n_filters_keep = growth_rate * nb_layers[nb_dense_block + block_idx]
 
-        if K.image_dim_ordering() == 'th':
+        if K.image_data_format() == 'channels_first':
             out_shape[1] = n_filters_keep
         else:
             out_shape[3] = n_filters_keep
 
         # upsampling block must upsample only the feature maps (concat_list[1:]),
-        # not the concatenation of the input with the feature maps (concat_list[0].
-        l = concatenate(concat_list[1:], axis=concat_axis)
+        # not the concatenation of the input with the feature maps
+        # (concat_list[0].
+        l = merge(concat_list[1:], mode='concat', concat_axis=concat_axis)
+        # l = concatenate(concat_list[1:], axis=concat_axis)
```

Comment:
Will be ready after https://github.com/fchollet/keras/pull/6035 is merged, which is likely quite soon. Also relevant https://github.com/fchollet/keras/pull/6034 and https://github.com/fchollet/keras/issues/5972. I might just wait a day or two instead of removing the comment. :-)